express-notes-api/index.js

```diff
@@ -0,0 +1,103 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json');
+const MiddleWare = express.json();
+app.use(MiddleWare);
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('Express server listening on port 3000');
+});
+// GET A LIST OF NOTES
+app.get('/api/notes/:id', (req, res) => {
+  const empty = [];
+  for (const id in data.notes) {
+    empty.push(data.notes[id]);
+  }
+  res.json(empty);
+});
+
+// GET SINGLE NOTE
+app.get('/api/notes/:id', (req, res) => {
+  const id = req.params.id;
+  if (parseInt(id) <= 0 || isNaN(id)) {
+    res.status(400);
+    res.json({ error: 'ID must be a positive int' });
+  } else if (data.notes[id]) {
+    res.json(data.notes[id]);
+  } else {
+    res.status(404);
+    res.json({ error: 'Unable to find the id' });
+  }
+});
+
+// CAN POST A NOTE
+const fs = require('fs');
+app.post('/api/notes/', (req, res) => {
+  if (req.body.content) {
+    res.status(400);
+    res.json({ error: 'Content not found' });
+  }
+  data.notes[data.nextId] = {};
+  data.notes[data.nextId].id = data.nextId;
+  data.notes[data.nextId].content = req.body.content;
+  data.nextId++;
+
+  fs.writeFile('./data.json', JSON.stringify(data, null, 2), err => {
+    if (err) {
+      // eslint-disable-next-line no-console
+      console.log(err);
+      res.status(500);
+      res.json({ error: 'Error' });
+    } else {
+      res.status(201);
+      res.json(data.notes[data.nextId - 1], { error: 'Success' });
```

Comment:
The `Response` object's `.json` method only expects one argument: https://expressjs.com/en/api.html#res.json